pkgs/development/interpreters/python/mk-python-derivation.nix

```diff
@@ -154,6 +154,7 @@ in
 , ... } @ attrs:
 
 assert (pyproject != null) -> (format == null);
+assert lib.assertMsg (format != "other") "${namePrefix}${attrs.pname}: `format = \"other\"` is deprecated. Use `pyproject = false` instead.";
```

Comment:
Might be better to keep `lib.warn` for a release, and change to assert/throw in the next.

Reply:
Since this change is just a replacement and the release included it is 25.05 in 6 months, I think we could throw an error.
We easily miss warnings, and I'm exhausted of pointing them out in reviews.

But if someone else also suggests `lib.warn`, I'll make the change.

Reply:
> Since these changes are just replacements, and the release included is 25.05 in 6 months, I think we could throw an error.

That will still be a show stopper for people using unstable channel. Such changes are not completely unacceptable, but we generally don't do this.

> We easily miss warnings, and I'm exhausted of pointing them out in reviews.

Ofborg will turn it into an error so such new patterns won't be accepted into Nixpkgs.

Reply:
Ok, let's change it to warn. I found the evaluation errors around home-assistant, so I'll commit it along with this fix.
Thanks for your review.